vendor/engines/workshops_engine/app/controllers/workshops_engine/application_controller.rb

```diff
@@ -7,12 +7,12 @@ class ApplicationController < ::ApplicationController
     private
 
     def restrict_access
-      redirect_to session[:last_moudle] unless
+      redirect_to main_app.user_path(current_user.id) unless
```

Comment:
all these controllers makes me think you would need to implement a better way to restrict access... I mean, maybe just creating some kind of module then extending the controllers with it, or making use of inheritance ... don't know, but surely something could be done (given the level of duplication you have)